### PR TITLE
Feat: Allow object in Grid prop cols, deprecate desktop and tablet #DS-995

### PR DESCRIPTION
**Pull request by crishpeen**
## Description

Deprecated desktop and tablet props.
Allow object inside cols prop.
Used the new cols prop object everywhere (hopefully).
Two small docs/test fixes.

### Additional context

Draft till next sprint.

### Issue reference

DS-995

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-twig/src/Resources/components/Grid/README.md [hunk 1]
```diff
@@ -196,3 +198,4 @@ see the [Escape hatches][escape-hatches] section in README to learn how and when
 [grid]: https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web/src/scss/components/Grid
 [escape-hatches]: https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web-twig/README.md#escape-hatches
 [digitalocean-span]: https://www.digitalocean.com/community/tutorials/css-css-grid-layout-span-keyword
+[deprecated]: https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web-twig/README.md#deprecations
```
Comment (pavelklibani) [inline packages/web-twig/src/Resources/components/Grid/README.md:201]: 🔤

**Review by pavelklibani — APPROVED** [review]

File: packages/web-react/src/components/Grid/README.md [hunk 2]
```diff
@@ -17,14 +17,14 @@ Use Grid to build multiple column layouts. This Grid works on twelve column syst
...
-| `elementType`      | HTML element                              | `div`   | ✕        | Element type to use for the Grid    |
-| `tablet`           | [`1` \| `2` \| `3` \| `4` \| `6` \| `12`] | —       | ✕        | Number of columns to use on tablet  |
-| `UNSAFE_className` | `string`                                  | —       | ✕        | Wrapper custom class name           |
-| `UNSAFE_style`     | `CSSProperties`                           | —       | ✕        | Wrapper custom style                |
+| Name               | Type                                                         | Default | Required | Description                                                                          |
+| ------------------ | ------------------------------------------------------------ | ------- | -------- | ------------------------------------------------------------------------------------ |
+| `cols`             | [`1` \| `2` \| `3` \| `4` \| `5` \| `6` \| `12` \| `object`] | —       | ✕        | Number of columns to use, use object to set responsive values                        |
```
Comment (literat) [inline packages/web-react/src/components/Grid/README.md:22]:
```suggestion
| `cols`             | [`1` \| `2` \| `3` \| `4` \| `5` \| `6` \| `12` \| `object`] | —       | ✕        | Number of columns to use, use object to set responsive values, e.g. `{ mobile: 1, tablet: 2, desktop: 3 }`                        |
```

I know that there is an example before this table, but in this case better to repeat it IMHO.

**Review by literat — APPROVED** [review]
LGTM 👏

File: packages/web-twig/src/Resources/components/Grid/__tests__/__fixtures__/gridDefault.twig
Comment (adamkudrna) [inline packages/web-twig/src/Resources/components/Grid/__tests__/__fixtures__/gridDefault.twig:1]: I'd like to see these test cases:

- number
- object with all breakpoints
- object with only some breakpoints

**Review by adamkudrna — APPROVED** [review]

**Pull request merged**